### DISCUSSION TITLE
Discord command not running

### DISCUSSION THREAD

**Discussion by changcheng967** [Rapptz/discord.py#9743] [General]
My code is like
`import discord
from discord.ext import commands
from peewee import SqliteDatabase, Model, IntegerField, CharField
import random

intents = discord.Intents(
    messages=True,          # Enable message content intent
    guilds=True,            # Enable the guilds intent
    members=True,           # Enable the members intent
    presences=True          # Enable the presences intent
)

# Create an instance of the bot
bot = commands.Bot(command_prefix='/', intents=intents)

# Discord bot token
TOKEN = 'my token'

# Set your database file name
DB_FILE = 'economy_database.db'

# Set the currency symbol
CURRENCY_SYMBOL = 'Dollars'

# database connection
db = SqliteDatabase(DB_FILE)

# Define the User and Item models
class User(Model):
    user_id = IntegerField(unique=True)
    balance = IntegerField(default=0)

    class Meta:
        database = db

class Item(Model):
    name = CharField(unique=True)
    price = IntegerField()

    class Meta:
        database = db

# Define the Lottery model
class Lottery(Model):
    user_id = IntegerField()
    ticket_number = IntegerField()

    class Meta:
        database = db

# Initialize the database
db.connect()
db.create_tables([User, Item, Lottery], safe=True)

@bot.event
async def on_message(message):
    print(f"Received message: {message.content}")

    if message.content.startswith('/'):
        print("Potential command detected.")

    # Process commands
    await bot.process_commands(message)

    # Ignore messages from the bot itself to prevent infinite loops
    if message.author == bot.user:
        return
    
@bot.command(name='/ping', help='Check bot latency')
async def ping(ctx):
    try:
        latency = round(bot.latency * 1000)
        await ctx.send(f'Pong! Latency: {latency}ms')
    except Exception as e:
        print(f"An error occurred in the ping command: {e}")

@bot.event
async def on_ready():
    print(f'Logged in as {bot.user.name} ({bot.user.id})')
    print('-------')
    await update_bot_status()

@bot.event
async def on_message(message):
    print(f"Received message: {message.content}")
    
    # Process commands
    await bot.process_commands(message)

    # Ignore messages from the bot itself to prevent infinite loops
    if message.author == bot.user:
        return

async def update_bot_status():
    await bot.change_presence(activity=discord.Game(name="made by changcheng967"))

@bot.command(name='balance', help='Check your balance')
async def balance(ctx):
    user = get_or_create_user(ctx.author.id)
    await ctx.send(f'Your balance: {user.balance} {CURRENCY_SYMBOL}')

@bot.command(name='shop', help='View available items in the shop')
async def shop(ctx):
    items = Item.select()
    shop_list = '\n'.join([f'{item.name} - {item.price} {CURRENCY_SYMBOL}' for item in items])
    await ctx.send(f'**Shop Items**:\n{shop_list}')

@bot.command(name='buy', help='Buy an item from the shop')
async def buy(ctx, item_name: str):
    user = get_or_create_user(ctx.author.id)
    item = get_item(item_name)

    if item and user.balance >= item.price:
        user.balance -= item.price
        user.save()
        await ctx.send(f'You bought {item.name} for {item.price} {CURRENCY_SYMBOL}! New balance: {user.balance} {CURRENCY_SYMBOL}')
    elif not item:
        await ctx.send('Item not found in the shop')
    else:
        await ctx.send('Insufficient funds')

@bot.command(name='sell', help='Sell an item to the shop')
async def sell(ctx, item_name: str):
    user = get_or_create_user(ctx.author.id)
    item = get_item(item_name)

    if item:
        user.balance += item.price
        user.save()
        await ctx.send(f'You sold {item.name} for {item.price} {CURRENCY_SYMBOL}! New balance: {user.balance} {CURRENCY_SYMBOL}')
    else:
        await ctx.send('Item not found in the shop')

@bot.command(name='lottery', help='Buy a lottery ticket')
async def lottery(ctx):
    user = get_or_create_user(ctx.author.id)
    if user.balance >= 50:  # Assuming lottery ticket price is 50 coins
        user.balance -= 50
        user.save()

        ticket_number = random.randint(1, 100)
        Lottery.create(user_id=user.user_id, ticket_number=ticket_number)

        await ctx.send(f'You bought a lottery ticket with number {ticket_number} for 50 {CURRENCY_SYMBOL}! Good luck!')

    else:
        await ctx.send(f'You need at least 50 {CURRENCY_SYMBOL} to buy a lottery ticket')

def get_or_create_user(user_id):
    user, created = User.get_or_create(user_id=user_id)
    return user

def get_item(item_name):
    try:
        item = Item.get(Item.name == item_name)
        return item
    except Item.DoesNotExist:
        return None

#buying

try:
    roblox_pass = Item.get(Item.name == 'Roblox Season Pass')
    roblox_pass.price = 10000000
    roblox_pass.save()

    minecraft_account = Item.get(Item.name == 'Minecraft Java Edition Account')
    minecraft_account.price = 50000000
    minecraft_account.save()

except Item.DoesNotExist:
    # Create items if they don't exist
    Item.create(name='Roblox Season Pass', price=10000000)
    Item.create(name='Minecraft Java Edition Account', price=50000000)

# Run the bot
bot.run(TOKEN)
`

but then the warning said WARNING  discord.ext.commands.bot Privileged message content intent is missing, commands may not work as expected. I tryed to fix it but it's still there, when I tried to use the command, there is no command when I type "/"

**Comment by No767** [comment]
@changcheng967 You should ideally reformat your code to utilize codeblocks. This is not an issue with discord.py first and foremost.

One notable issue that I want to discuss is the usage of passing `/` into the keyword `command_prefix` found in `commands.Bot`. There is the claims of by passing in `/` to the keyword `command_prefix`, that Discord would magically recognize it and handle accordingly. **This is purely false**. App commands have to be sent to Discord and synced. I would strongly recommend you see the [`app_commands/basic`](https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py) in order to utilize app commands (aka slash commands) in an proper manner with discord.py.
